tests/Microsoft.DotNet.Framework.Docker.Tests/DockerHelper.cs

```diff
@@ -60,26 +61,50 @@ private void Execute(string args)
                 string msg = $"Failed to execute {startInfo.FileName} {startInfo.Arguments}{Environment.NewLine}{stdError}";
                 throw new InvalidOperationException(msg);
             }
+
+            return result;
         }
 
         public static bool ImageExists(string tag)
         {
-            ProcessStartInfo startInfo = new ProcessStartInfo("docker", $"image ls -q {tag}");
-            startInfo.RedirectStandardOutput = true;
-            Process process = Process.Start(startInfo);
-            string stdOutput = process.StandardOutput.ReadToEnd().Trim();
-            process.WaitForExit();
-            return process.ExitCode == 0 && stdOutput != "";
+            return VerifyImageStatus(tag, "image ls -q");
         }
 
-        public void Run(string image, string containerName, string command)
+        public void Run(string image, string containerName, string command, bool detach = false)
         {
-            Execute($"run --rm --name {containerName} {image} {command}");
+            string options = detach ? "run --rm -d --name" : "run --rm --name";
+            Execute($"{options} {containerName} {image} {command}");
         }
 
         public void Pull(string image)
         {
             Execute($"pull {image}");
         }
+        public void Stop(string containerName)
+        {
+            Execute($"stop {containerName}");
+        }
+
+        public string GetContainerAddress(string container)
+        {
+            string address =  Execute("inspect -f \"{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}\" " + container);
+            //remove the last character of the address
+            return address.Remove(address.Length - 1);
+        }
+        public static bool ImageRunning(string tag)
+        {
+            return VerifyImageStatus(tag, "inspect -f '{{.State.Running}}'");
+        }
+
+        private static bool VerifyImageStatus(string tag, string command)
```

Comment:
The name of this method is rather ambiguous to me.  The method itself isn't technically tied to `images`.

Reply:
Switched to Container, what do you think?

Reply:
This still doesn't make sense.  The method as written is abstract and is being used for both methods and containers.  A concrete point, is the tag parameter is confusing when checking to see if a container exists.